src/global/global_init.cc

```diff
@@ -261,6 +261,11 @@ int global_init_prefork(CephContext *cct, int flags)
 {
   if (g_code_env != CODE_ENVIRONMENT_DAEMON)
     return -1;
+    
+  if (test_pid_file_in_use(g_conf) < 0) {
```

Comment:
before lock, it will try to open the file with O_CREAT flag. so if empty, pid_file creates.